openslide/__init__.py

```diff
@@ -226,6 +237,18 @@ def read_region(self, location, level, size):
             self._osr, location[0], location[1], level, size[0], size[1]
         )
 
+    def set_cache(self, cache):
+        """Use the specified cache to store recently decoded slide tiles.
+
+        By default, the object has a private cache with a default size.
+
+        cache: an OpenSlideCache object."""
+        try:
+            llcache = cache._openslide_cache
```

Comment:
I see you are using a hand-crafted type checking. At some point we probably should switch to type annotations, although I don't think those get enforced so a runtime check like this might still be needed even with typing.

Reply:
This wasn't really intended as type-checking, and I'm also not thrilled by the private member access.  I didn't have a better idea for retrieving the low-level cache handle; any thoughts?

You're right that we should add type hints.  Filed that as #157.

Reply:
not unless you add an accessor function, but that would make the internals more explicit and might tie us down if we want to change the cache api later.

I can imagine a future implementation of the cache might use a callback function so someone could add an external memcached or redis process to share a cache between processes.